src/includes/performance/concurrency/native.mdx

```diff
@@ -34,3 +34,17 @@ Following up on that warning, `sentry_transaction_context_set_name()`, which ope
  */
 
 ```
+
+Those familiar with performance monitoring on other platforms may note that scopes are one of the tools to work around concurrency. This is not the case for the native SDK, and is discussed in the following two sections. Expectations for users adding manual instrumentation to their app that are exclusive to the native SDK due to scopes are also detailed below.
+
+### Single Scope, Multiple Transactions
+
+The native SDK makes a major compromise in an attempt to follow a server-like model despite having a single scope: A user may create multiple concurrent transactions and multiple concurrent spans, at the cost of making it their responsibility to manage the transactions and spans. It is on them to ensure that no unsafe mutations are being made to transactions while they are being used. In comparison, other SDKs either follow a single (hidden) scope, single transaction model where all spans across multiple threads are collapsed into one single transaction.
```

Comment:
Initially I was hesitant on the amount that was cut out from here (I'd actually added even _more_ while addressing the comment prior to this because I realized that the last sentence in the original paragraph just left even more questions).

Having mulled on this a bit though, I don't think this document is the best place to be explaining design decisions - it's supposed to just guide you into getting started with tracing using the native SDK. A full explanation belongs maybe at the bottom of this doc as bonus content if you're curious as to why things are the way they are. This would allow us to write this paragraph so it focuses primarily on what the end user needs to do, which I think makes it really easy to just get to the point, to put it crudely.

What are your thoughts on this, particularly to the idea of moving these sections to the bottom of the document?

Reply:
Performance monitoring in the native SDK comes with very little hand-holding partially due to it being a bare bones implementation, but also partially due to the language it's written in. The out of box experience, to put it metaphorically comes with several guns to shoot your developer limbs off with; The sections make an attempt to at least warn the user about some of these guns and to handle them with care. Worst case scenario, it's possible for a developer's application to crash if they bump into one of these issues.

I'll see what I can do to just integrate the most important parts of these new bits into earlier sections.

Reply:
That makes sense to me. I think warning users at the spots where they need to be warned is the right approach vs adding a lot more explanation about why the warning is there.

Reply:
Alright, sorry about all of the churn on this PR: the diff should be reviewable. I've removed all of the new bits from this file and added whatever makes sense to `src/includes/performance/connect-errors-spans/native.mdx`  instead, so that should be the only thing that needs to be revisited. @imatwawana could I bug you to take a final look at that file? It's relatively short, ~2 sentences (if that helps).

Regarding the reasoning behind the changes, which I'm pretty sure my teammates will be curious about: The new diff makes more sense this way anyway as scopes were introduced **after** the section that I originally updated (concurrency/threading concerns). This meant that it was talking about a concept that the docs hadn't explained yet. The new file explains scopes, which makes it a natural place to talk about how they work in native.

I've also opted to remove the section talking about child spans automatically binding themselves to the scope as the auto-binding behaviour could only be found in the python SDK. It doesn't look like JS, Cocoa, nor Ruby do this, so it isn't too meaningful to note how the native SDK is different from just one other SDK.

Reply:
Looks good to me! Thanks for all your work on this! And yeah, I suspect the Vercel preview will work again next time you commit changes. It's finicky sometimes.